### DISCUSSION TITLE
Facing issue when upgrading versions

### DISCUSSION THREAD

**Discussion by uttam-gelot** [pact-foundation/nestjs-pact#80] [Q&A]
current versions: of `@pact-foundation/pact` is `9.18.1` and `nestjs-pact` is `2.2.1`. now when I try to update the versions, as per mentioned in the [documentation](https://github.com/pact-foundation/nestjs-pact#:~:text=Pact%2DJS%20v10.x%2B%20users), when I run `npm i -D nestjs-pact @pact-foundation/pact`. it updates the version of `@pact-foundation/pact` to`13.1.2` and `nestjs-pact` to `2.3.1`. 

But now all the test case are failing and getting errors. before mentioning error, I'll show current code setup.

`pact.module.ts` file looks like below.
![image](https://github.com/user-attachments/assets/7da5e959-c3ad-4228-b6fe-a6b4f2fbb56e)

and consumer test file looks like below
![image](https://github.com/user-attachments/assets/1a193878-ee90-439c-bf29-7b3618782b28)

now when I run test then get below errors.
![image](https://github.com/user-attachments/assets/b3122af0-9def-4acd-88d0-04cd522357c7)

**Comment by uttam-gelot** [comment]
@YOU54F could you please help me to figure this out?

**Comment by YOU54F** [comment]
nestjs pact has not been updated to use the latest 12.x+ versions of pact-js i'm afraid.

13.x removes the publisher interface https://github.com/pact-foundation/pact-js/releases/tag/v13.0.0

See PR where someone attempted to start support for the v3 interface

https://github.com/pact-foundation/nestjs-pact/pull/66

**Reply by uttam-gelot** [reply]
then which is the latest version, which I can use? @YOU54F

**Reply by uttam-gelot** [reply]
> nestjs pact has not been updated to use the latest 12.x+ versions of pact-js i'm afraid.

if we're not supporting latest version then I guess we should update `npm i -D nestjs-pact @pact-foundation/pact` line with version which we support. just like what we did for `npm i -D nestjs-pact@2.2.2 @pact-foundation/pact@9.18.1`

**Comment by uttam-gelot** [comment]
@YOU54F  When tried `@pact-foundation/pact` with version `10.4.0` then I'm getting below error.
![image](https://github.com/user-attachments/assets/9781d934-2182-4868-a0dc-bd43e11ccda1)

**Comment by YOU54F** [comment]
Hey this should now work with nest-js v10, and the latest pact-js (currently v13)

examples consumer and provider project in the below link

https://github.com/YOU54F/nestjs-pact-test/tree/main/v2_spec/pactjs_v10_nestjspact_v3

**Reply by uttam-gelot** [reply]
Thank you @YOU54F but I'd already fixed this issue.
